docs/data/introduction/licensing/licensing.md

```diff
@@ -281,6 +281,11 @@ However, when the term ends, you are not allowed to use the current or older ver
 
 To solve the issue, you can [renew your license](https://mui.com/r/x-get-license/) or stop making changes to code depending on MUI X's APIs.
 
+:::warning
+Make sure to set `process.env.NODE_ENV` to `'production'` in your build process to avoid the watermark in production.
+Most bundlers set this environment variable automatically when building for production, but for custom setups, you might need to set it manually.
```

Comment:
It would probably be good to expand on this or reference some docs on `why` it is not a good idea to have `NODE_ENV` set to anything else than `production` when the code is live. 

I've been to plenty of companies where `NODE_ENV=stage` or `NODE_ENV=prod` were used and I had to go on a crusade to convince people who didn't know about this. 🙃

Reply:
Do you have any good links we could share with our users here, @JCQuintas?

Reply:
I agree that if we can make people understand that it's not some MUI specific needs but that you should always respect this convention because lots of packages relies on this infomation

Reply:
Not really, this is a convention only... Hence there is no actual documentation from node on it. 

Maybe [webpack](https://webpack.js.org/guides/production/#specify-the-mode)? Though it is not super complete. I would try to inform people why it is a bad idea and link to a few similar sources.

@flaviendelangle yeah, but you can't expect everyone to be reasonable about this. If we have actual documentation on the `why`, then an engineer can go into their superiors and say `"this pattern we are using is wrong because it can cause issues in downstream libraries, here is the docs"`.

It prevents issues on our side, if someone comes in asking `"can you remove the watermark when NODE_ENV=prod?"` and it prevents issues on their side, because it is not a "whim" of the frontend engineer, it is an actual issue that the `DevOps|Backend|Frontend` teams need to care about.

Reply:
>  If we have actual documentation on the why, then an engineer can go into their superiors and say "this pattern we are using is wrong because it can cause issues in downstream libraries, here is the docs".

I totally agree with you :ok_hand: 

My point is that if this message can be understood as "you should not do it because it's bad" and not "you sould not do it because otherwise our product breaks :cry: " it's better :laughing:

Reply:
Updated 👍🏻
WDYT?

<img width="837" src="https://github.com/user-attachments/assets/6a7c7998-9a9e-4c83-adae-ea7bed44a929">